wagtail/blocks/field_block.py

```diff
@@ -816,10 +818,21 @@ def bulk_to_python(self, values):
 
         The instances must be returned in the same order as the values and keep None values.
         """
-        objects = self.model_class.objects.in_bulk(values)
-        return [
-            objects.get(id) for id in values
-        ]  # Keeps the ordering the same as in values.
+        uuid_pattern = re.compile(
+            r"^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$"
```

Comment:
I am not sure this is the right approach, we are creating a very fragile check for if the item is a uuid object.

Why can't we do something simpler like this?


```python
str_values = [str(value) for value in values]
return [ str(objects.get(id)) for id in str_values ]
```

This resolves the potential object-like items into strings, then compares them to the values, also converted to strings.

Reply:
Thanks, @lb-. I've already tried to use this approach, but `str(objects.get(id))` is returning 'None'. I believe the current issue arises when we use `objects.get(id)` with UUID primary keys, as the objects resembles a dictionary like this:

`{UUID('21c6d0a9-28c8-4b56-841c-cef373619f2e'): <MySnippet: MySnippet object (21c6d0a9-28c8-4b56-841c-cef373619f2e)>}`

Consequently, when we use `str(objects.get(id))`, it tries to retrieve the value with a key `(e.g., objects.get('21c6d0a9-28c8-4b56-841c-cef373619f2e'))`, resulting in None being returned.

To resolve this, we need to use `**objects.get(UUID(id))**`, which cannot be used with regular primary keys. This led me to add a check to differentiate between UUID and integer primary keys.

Do you have any suggestions on how to handle this situation in a better way? Please correct me if I've misunderstood anything!